rest-client/src/main/java/org/eclipse/dataspaceconnector/registration/client/ApiClientFactory.java

```diff
@@ -19,18 +19,26 @@
 import org.eclipse.dataspaceconnector.spi.result.Result;
 import org.jetbrains.annotations.NotNull;
 
+import java.time.Duration;
 import java.util.function.Function;
 
+import static org.eclipse.dataspaceconnector.common.configuration.ConfigurationFunctions.propOrEnv;
+
 /**
  * Factory class for {@link ApiClient}.
  */
 public class ApiClientFactory {
+
+    private static final int API_CLIENT_CONNECT_TIMEOUT = Integer.parseInt(propOrEnv("API_CLIENT_CONNECT_TIMEOUT", "30"));
```

Comment:
Any reason why this should not be a normal EDC config value? those are overridable by environment variables.
also, name should be a constant and annotated with `@EdcSetting` with the appropriate attributes.

Reply:
No particular reason, just didn't do it before thinking that as this is for rest client only (Which is generated from open api) and not related to an edc extension.

But now updated it as per your suggestion. So please take a look.